utils/deploy.ts

```
@@ -109,113 +121,93 @@ export const configureDeploymentSettings = async ({
+  return hre.deployOrUpgradeProxy<Certificate, Certificate__factory>({
+    contractName: 'Certificate',
+    args: [],
+    options: { initializer: 'initialize()' },
+  });
+};
+
+export const deployFIFOMarketContract = async ({
+  hre,
+  feeWallet,
+  feePercentage,
+}: {
+  hre: CustomHardHatRuntimeEnvironment;
+  feeWallet: Address;
+  feePercentage: number;
+}): Promise<InstanceOfContract<FIFOMarket>> => {
+  const deployments = await hre.deployments.all();
+  return hre.deployOrUpgradeProxy<FIFOMarket, FIFOMarket__factory>({
+    contractName: 'FIFOMarket',
+    args: [
+      deployments['Removal']!.address,
+      deployments['BridgedPolygonNORI']!.address,
+      deployments['Certificate']!.address,
+      feeWallet,
+      feePercentage,
+    ],
+    options: {
+      initializer: 'initialize(address,address,address,address,uint256)',
+    },
+  });
+};
+
+export const deployBridgedPolygonNORIContract = async ({
+  hre,
+  childChainManagerProxyAddress,
+}: {
+  hre: CustomHardHatRuntimeEnvironment;
+  childChainManagerProxyAddress: Address;
+}): Promise<InstanceOfContract<BridgedPolygonNORI>> => {
+  return hre.deployOrUpgradeProxy<
+    BridgedPolygonNORI,
+    BridgedPolygonNORI__factory
+  >({
+    contractName: 'BridgedPolygonNORI',
+    args: [childChainManagerProxyAddress],
+    options: { initializer: 'initialize(address)' },
+  });
+};
+
+export const deployNORIContract = async ({
+  hre,
+}: {
+  hre: CustomHardHatRuntimeEnvironment;
+}): Promise<InstanceOfContract<NORI>> => {
+  return hre.deployOrUpgradeProxy<NORI, NORI__factory>({
+    contractName: 'NORI',
+    args: [],
+  });
+};
+
+export const deployLockedNORIContract = async ({
+  hre,
+}: {
+  hre: CustomHardHatRuntimeEnvironment;
+}): Promise<InstanceOfContract<LockedNORI>> => {
+  return hre.deployOrUpgradeProxy<LockedNORI, LockedNORI__factory>({
+    contractName: 'LockedNORI',
+    args: [(await hre.deployments.get('BridgedPolygonNORI'))!.address],
```

should we throw an error if it's not found instead (instead of the non-null assertion)? If you try to pass it `undefined` it will cost gas + throw anyways